### ISSUE TITLE
[HDX-36] feat: support Histograms (metrics) backend API

### ISSUE THREAD

**Issue by wrn14897** [hyperdxio/hyperdx#48]
Currently histogram is ingested but not shown. Need to implement it on the api side and make sure it returns the json data that is renderable

<sub>From [SyncLinear.com](https://synclinear.com) | [HDX-36](https://linear.app/hyperdx/issue/HDX-36/feat-support-histograms-metrics-backend-api)</sub>

**Comment by jaggederest** [comment]
Pausing this for now until user feedback and strategy are gathered 👍

**Comment by MikeShi42** [comment]
Since the user request is as we predicted returning the X percentile of latency given a histogram, it looks like we need to do some math.

Regarding how we should do the math, Prometheus actually published a pretty decent article on this and the drawbacks of summary metrics that we talked about last week: [https://prometheus.io/docs/practices/histograms/](https://prometheus.io/docs/practices/histograms/)

The tl;dr is that they indeed linearly interpolate within a bucket as we suspected, not sure how we're going to convert that logic into sql 😂

**Comment by wrn14897** [comment]
query implementation https://github.com/hyperdxio/hyperdx/pull/253
